sdk/servicebus/service-bus/samples/javascript/advanced/deferral.js

```diff
@@ -75,20 +59,20 @@ async function sendMessages() {
   }
   // wait until all the send tasks are complete
   await Promise.all(promises);
-  await queueClient.close();
+  await sender.close();
   await sbClient.close();
 }
 
 async function receiveMessage() {
-  const sbClient = ServiceBusClient.createFromConnectionString(connectionString);
+  const sbClient = new ServiceBusClient(connectionString);
 
-  // If receiving from a Subscription, use `createSubscriptionClient` instead of `createQueueClient`
-  const queueClient = sbClient.createQueueClient(queueName);
+  // If receiving from a subscription you can use the getReceiver(topic, subscription) overload
+  let receiver = sbClient.getReceiver(queueName, "peekLock");
```

Comment:
Side note not related to this PR: It just struck me that we no longer have an enum and users are expected to use string literals which is easy to mess up via typos :(

Reply:
It's a discriminated union with two values so it shows up in auto complete (and we do runtime checking).

Reply:
(if that alleviates your concern)